src/Centreon/Infrastructure/HostConfiguration/Repository/HostMacroRepositoryRDB.php

```diff
@@ -63,4 +64,49 @@ public function addMacroToHost(Host $host, HostMacro $hostMacro): void
         $hostMacroId = (int)$this->db->lastInsertId();
         $hostMacro->setId($hostMacroId);
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function findAllByHost(Host $host): array
+    {
+        Assertion::notNull($host->getId(), 'Host::id');
+        $statement = $this->db->prepare(
+            $this->translateDbName('SELECT * FROM `:db`.on_demand_macro_host WHERE host_host_id = :host_id')
+        );
+        $statement->bindValue(':host_id', $host->getId(), \PDO::PARAM_INT);
+        $statement->execute();
+        $hostMacros = [];
+        while ($result = $statement->fetch(\PDO::FETCH_ASSOC)) {
+            $hostMacros[] = HostMacroFactoryRdb::create($result);
+        }
+        return $hostMacros;
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function updateMacro(HostMacro $hostMacro): void
+    {
+        Assertion::notNull($hostMacro->getId(), 'HostMacro::id');
+        Assertion::notNull($hostMacro->getHostId(), 'HostMacro::host_id');
+        $statement = $this->db->prepare(
+            $this->translateDbName(
+                'UPDATE `:db`.on_demand_macro_host
+                    SET host_macro_name = :new_name,
+                        host_macro_value = :new_value,
+                        is_password = :is_password,
+                        description = :new_description,
+                        macro_order = :new_order
+                WHERE host_macro_id = :id'
+            )
+        );
+        $statement->bindValue(':new_name', $hostMacro->getName());
+        $statement->bindValue(':new_value', $hostMacro->getValue());
+        $statement->bindValue(':is_password', $hostMacro->isPassword() ? '1' : null);
```

Comment:
is_password can be null ?

Reply:
Yes but I think it's not a good thing to put a null value instead of specifying a boolean value.
I will fix it.